scheduler/import_handlers.py

```diff
@@ -1,94 +1,164 @@
+        professor_data = row[5].split(" ")
+        if len(professor_data) == 2:
+            start_time = datetime.timedelta(hours=row[2].hour, minutes=row[2].minute)
+            end_time = datetime.timedelta(hours=row[3].hour, minutes=row[3].minute)
+            start_date = row[1].to_pydatetime() + start_time
+            end_date = row[1].to_pydatetime() + end_time
+            if start_date > end_date:
+                incorrect.append(row[0])
+                continue
+            professor = get_professor(professor_data[0], professor_data[1])
+            auditorium = get_auditorium(str(row[7]))
+            group = get_group(row[6])
+            _lesson, created = Lesson.objects.get_or_create(
+                name=row[4],
+                professor=professor,
+                auditorium=auditorium,
+                group=group,
+                start_time=start_date,
+                end_time=end_date
+            )
+            if created:
                 added_columns += 1
-    return added_columns
+            else:
+                duplicate.append(row[0])
+        else:
+            incorrect.append(row[0])
+    return added_columns, incorrect, duplicate
 
 
-def save_into_database(parameters: tuple) -> bool:
-    """
-    Save a lesson (and other model if necessary) objects into database
-    :param parameters: tuple of formatted data needed for database:
-                (lesson_name, prof_name, prof_surname, auditorium, group, start_date, end_date)
-    :return: whether given lesson already existed or not
-    """
-    print(parameters)
-    professor, _created = Professor.objects.get_or_create(
-        name=parameters[1],
-        surname=parameters[2]
-    )
-    auditorium, _created = Auditorium.objects.get_or_create(
-        number=parameters[3]
-    )
-    group, _created = Group.objects.get_or_create(
-        number=parameters[4]
-    )
-    _lesson, created = Lesson.objects.get_or_create(
-        name=parameters[0],
-        professor=professor,
-        auditorium=auditorium,
-        group=group,
-        start_time=parameters[5],
-        end_time=parameters[6]
-    )
-    return created
+def check_types_csv(row: tuple) -> bool:
+    """Returns true if row from csv file has correct types"""
+    if not isinstance(row[1], str):
+        return False
+    if not isinstance(row[2], str) and isinstance(row[3], str):
+        return False
+    if not isinstance(row[4], str) and isinstance(row[5], str):
+        return False
+    try:
+        int(row[6])
+    except ValueError:
+        return False
+    if not isinstance(row[7], (str, int, float)):
+        # 3.27, 3.27a and 137 should all be supported
+        return False
+    return True
+
+
+def check_types_excel(row: tuple) -> bool:
+    """Returns true if row from excel file has correct types"""
+    if not isinstance(row[1], pd.Timestamp):
+        return False
+    if not (isinstance(row[2], datetime.time) and isinstance(row[3], datetime.time)):
```

Comment:
Co jest w row[3] że **nie powinno** być timem?

Reply:
Jeśli nie jest prawdą że row[2] i row[3] jest timem to źle?